File: trax_app/lib/models/bike_model.dart
class BikeModel {
  final int id;
  final String modelName;
  final String? motorType;
  final String? controller;
  final String? imageUrl;
  final String brandName;
  final int? motorPeakPowerW;
  final int? motorTorqueNm;
  final String? batteryType;
  final String? batteryVoltage;
  final int? batteryCapacityWh;

  BikeModel({
    required this.id,
    required this.modelName,
    this.motorType,
    this.controller,
    this.imageUrl,
    required this.brandName,
    this.motorPeakPowerW,
    this.motorTorqueNm,
    this.batteryType,
    this.batteryVoltage,
    this.batteryCapacityWh,
  });

  factory BikeModel.fromJson(Map<String, dynamic> json) {
    return BikeModel(
      id: json['id'] as int,
      modelName: json['modelName'] as String,
      motorType: json['motorType'] as String?,
      controller: json['controller'] as String?,
      imageUrl: json['imageUrl'] as String?,
      brandName: json['brandName'] as String,
      motorPeakPowerW: json['motorPeakPowerW'] as int?,
      motorTorqueNm: json['motorTorqueNm'] as int?,
      batteryType: json['batteryType'] as String?,
      batteryVoltage: json['batteryVoltage'] as String?,
      batteryCapacityWh: json['batteryCapacityWh'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'modelName': modelName,
      'motorType': motorType,
      'controller': controller,
      'imageUrl': imageUrl,
      'brandName': brandName,
      'motorPeakPowerW': motorPeakPowerW,
      'motorTorqueNm': motorTorqueNm,
      'batteryType': batteryType,
      'batteryVoltage': batteryVoltage,
      'batteryCapacityWh': batteryCapacityWh,
    };
  }
}

File: backend/src/main/java/com/trax/model/BikeModel.java
package com.trax.model;

import jakarta.persistence.*;

@Entity
@Table(name = "bike_model")
public class BikeModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "brand_id", nullable = false)
    private BikeBrand brand;

    @Column(nullable = false)
    private String modelName;

    @Column(length = 500)
    private String imageUrl;

    private String motorType;
    @Column(name = "motor_peak_power_w")
    private Integer motorPeakPowerW;
    private Integer motorTorqueNm;
    private String controller;
    private String batteryType;
    private String batteryVoltage;
    private Integer batteryCapacityWh;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public BikeBrand getBrand() { return brand; }
    public void setBrand(BikeBrand brand) { this.brand = brand; }
    public String getModelName() { return modelName; }
    public void setModelName(String modelName) { this.modelName = modelName; }
    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }
    public String getMotorType() { return motorType; }
    public void setMotorType(String motorType) { this.motorType = motorType; }
    public Integer getMotorPeakPowerW() { return motorPeakPowerW; }
    public void setMotorPeakPowerW(Integer motorPeakPowerW) { this.motorPeakPowerW = motorPeakPowerW; }
    public Integer getMotorTorqueNm() { return motorTorqueNm; }
    public void setMotorTorqueNm(Integer motorTorqueNm) { this.motorTorqueNm = motorTorqueNm; }
    public String getController() { return controller; }
    public void setController(String controller) { this.controller = controller; }
    public String getBatteryType() { return batteryType; }
    public void setBatteryType(String batteryType) { this.batteryType = batteryType; }
    public String getBatteryVoltage() { return batteryVoltage; }
    public void setBatteryVoltage(String batteryVoltage) { this.batteryVoltage = batteryVoltage; }
    public Integer getBatteryCapacityWh() { return batteryCapacityWh; }
    public void setBatteryCapacityWh(Integer batteryCapacityWh) { this.batteryCapacityWh = batteryCapacityWh; }
}
